File: php/features.php
<?php
declare(strict_types=1);

// HealthcareGovContent SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class HealthcareGovContentFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new HealthcareGovContentBaseFeature();
            case "test":
                return new HealthcareGovContentTestFeature();
            default:
                return new HealthcareGovContentBaseFeature();
        }
    }
}

File: py/features.py
# HealthcareGovContent SDK feature factory

from feature.base_feature import HealthcareGovContentBaseFeature
from feature.test_feature import HealthcareGovContentTestFeature


def _make_feature(name):
    features = {
        "base": lambda: HealthcareGovContentBaseFeature(),
        "test": lambda: HealthcareGovContentTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
